swiss_locator/core/filters/swiss_locator_filter.py

```diff
@@ -498,6 +494,27 @@ def triggerResult(self, result: QgsLocatorResult):
                     QgsProject.instance().addMapLayer(ch_layer, False)
                     root.insertLayer(-1, ch_layer)
 
+                if empty_project:
+                    QCoreApplication.processEvents()  # wait for application to process pending tasks
+
+                    if (
+                        self.map_canvas.mapSettings().destinationCrs().authid()
+                        == "EPSG:3857"
+                    ):
+                        # Set the Swiss extent in EPSG:3857
+                        extent = QgsRectangle(624991, 5725825, 1209826, 6089033)
+                    else:  # Transform a WGS84 extent to map CRS
+                        bottom_left = QgsPoint(5.8, 45.2)
+                        top_right = QgsPoint(10.569, 48.322)
+                        bottom_left.transform(self.transform_4326)
+                        top_right.transform(self.transform_4326)
+                        extent = QgsRectangle(
+                            QgsPointXY(bottom_left), QgsPointXY(top_right)
+                        )
```

Comment:
@3nids, we could probably get rid of the `else` branch, since the VT are always loaded in `EPSG:3857`. Not sure if that'll change in the future or if any other VT could be loaded in another CRS. In that case, it'll make sense to keep it. 

What do you think?

Reply:
Avoided.